include/libdnf/base/goal_elements.hpp

```diff
@@ -156,10 +163,21 @@ struct GoalJobSettings : public ResolveSpecSettings {
     /// @exception libdnf::AssertionError When a different value already stored or when invalid value
     /// @since 1.0
     bool resolve_clean_requirements_on_remove();
+    /// Resolve upgrade_minimal value and store the result as the value used.
+    /// During advisory upgrades upgrade_minimal controls whether we upgrade to the minimal version
+    /// of a pkg that fixes given advisory or to the latest available version of the pkg (that includes the fix).
+    /// E.g., when pkg A-1 satistifes (fixes) an advisory and we have A-1, A-2 and A-3 available we upgrade
+    /// to A-3 by default but when upgrade-minimal is set we upgrade just to A-1.
+    ///
+    /// @return Resolved value.
+    /// @exception libdnf::AssertionError When a different value already stored or when invalid value
+    /// @since 5.0
+    bool resolve_upgrade_minimal();
 
     GoalUsedSetting used_strict{GoalUsedSetting::UNUSED};
     GoalUsedSetting used_best{GoalUsedSetting::UNUSED};
     GoalUsedSetting used_clean_requirements_on_remove{GoalUsedSetting::UNUSED};
+    GoalUsedSetting used_upgrade_minimal{GoalUsedSetting::UNUSED};
```

Comment:
Why do you want to implement it as goal setting rather then a new goal action?

Reply:
I didn't realize that is an option.

Looking at it now, that would mean two new action `upgrade_minimal` and `upgrade_all_minimal`?

Reply:
Yes, it is the option.

Reply:
I have changed it to goal actions instead of the goal settings. 
Please take a look.